plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/PublishApk.kt

```diff
@@ -24,8 +25,13 @@ open class PublishApk : PlayPublishPackageBase() {
     @get:PathSensitive(PathSensitivity.RELATIVE)
     @get:InputFiles
     internal val inputApks by lazy {
-        // TODO: If we take a customizable folder, we can fix #233, #227
-        variant.outputs.filterIsInstance<ApkVariantOutput>().map { it.outputFile }
+        if (extension.buildInputFolder == null) {
+            variant.outputs.filterIsInstance<ApkVariantOutput>().map { it.outputFile }
+        } else {
+            extension.buildInputFolder!!.listFiles(ApkFileFilter).toList().apply {
+                require(isNotEmpty(), { "Build input folder provided, but no APK files found" })
+            }
+        }
```

Comment:
We should never intentionally throw exceptions in tasks IMO. You know what they say, exceptions are for exceptional conditions. 😆 Anyway, I have it set up so Gradle will show a `NO-SOURCE` thing if the folder is empty which is the established convention. So your code should look something like this:
```kt
val override = extension.buildInputFolder
@Suppress("IfThenToElvis") // listFiles is nullable so the elvis would be wrong
if (override == null) {
    variant.outputs.filterIsInstance<ApkVariantOutput>().map { it.outputFile }
} else {
    override.listFiles(ApkFileFilter).orEmpty().toList()
}
```

Reply:
Looks interesting. I agree we should show `NO-SOURCE` if the input folder is empty. However, there's still a case where that input folder doesn't even exist. listFiles() would be null and we'd throw a null pointer exception. That shouldn't happen. For that case we should either

- throw a custom exception with a proper message
- log warning and show `NO-SOURCE`